### PR TITLE
Upgrade errorprone to 2.12.1 (from 2.11.0)

### PR DESCRIPTION
**Pull request by carterkozak**
==COMMIT_MSG==
Upgrade errorprone to 2.12.1 (from 2.11.0)
==COMMIT_MSG==

### Release notes:
https://github.com/google/error-prone/releases/tag/v2.12.0
https://github.com/google/error-prone/releases/tag/v2.12.1

### New checks
- [BoxedPrimitiveEquality](http://errorprone.info/bugpattern/BoxedPrimitiveEquality)
- [DoubleBraceInitialization](http://errorprone.info/bugpattern/DoubleBraceInitialization)
- [IgnoredPureGetter](http://errorprone.info/bugpattern/IgnoredPureGetter)
- [LockOnBoxedPrimitive](http://errorprone.info/bugpattern/LockOnBoxedPrimitive)
- [IncorrectMainMethod](http://errorprone.info/bugpattern/IncorrectMainMethod)
- [LongDoubleConversion](http://errorprone.info/bugpattern/LongDoubleConversion)
- [RobolectricShadowDirectlyOn](http://errorprone.info/bugpattern/RobolectricShadowDirectlyOn)
- [StaticAssignmentOfThrowable](http://errorprone.info/bugpattern/StaticAssignmentOfThrowable)
- [UnnecessaryLongToIntConversion](http://errorprone.info/bugpattern/UnnecessaryLongToIntConversion)
- [Varifier](http://errorprone.info/bugpattern/Varifier)

### CODE REVIEW

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java [hunk 1]
```diff
@@ -173,7 +173,9 @@ private static void configureErrorProneOptions(
                 "CatchSpecificity",
                 "InlineMeSuggester",
                 "PreferImmutableStreamExCollections",
-                "UnusedVariable");
+                "UnusedVariable",
+                // We don't use 'var' yet
```
Comment (schlosna) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java]:
```suggestion
                // We don't use 'var' yet, see VarUsage
```

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java [hunk 2]
```diff
@@ -173,7 +173,9 @@ private static void configureErrorProneOptions(
...
+                "Varifier");
```
Comment (schlosna) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java:178]: more explicitly, https://github.com/palantir/gradle-baseline/blob/develop/baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/VarUsage.java throws `error: [VarUsage] The `var` keyword results in illegible code in most cases and should not be used.` so conflicts with `Varifier`

**Review by robert3005 — APPROVED** [review]

**Review by schlosna — COMMENTED** [review]
[OSS CircleCI is having a bad Wednesday](https://status.circleci.com/)

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java (hunk 2 above)
Comment (carterkozak) [reply gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java:178]: :+1: good call, I've described the rationale in the comment

**Pull request merged**